algosdk/future/transaction.py

```diff
@@ -273,6 +271,13 @@ def undictify(d):
     def __eq__(self, other):
         if not isinstance(other, (Transaction, transaction.Transaction)):
             return False
+        if isinstance(other, transaction.Transaction):
```

Comment:
Added a warning for people who might try to compare the old `Transaction` object to a `future.Transaction` object